src/logic/calculate.js

```diff
@@ -0,0 +1,19 @@
+import operate from './operate';
+
+const calculate = ({ data, buttonVal }) => {
```

Comment:
- Here you are destructuring a single argument object, and then further down you destructure one of the properties of the original argument, so in order for this code to work the argument passed would have to be something like this:
`{data: {result: null, nextVal: null, operation: null}, buttonVal: 'AC'}`, while it might be correct it is a bit unnecessary and could lead to overcomplicating the app when you try to implement it in the UI, as well as the pathwright states that we need to pass two arguments, a data object and a buttonVal argument which might come in the form of a string.
Now, If in turn, we change the arguments to `const = (data, buttonVal)` we will make this code more accessible when we try to implement it from the UI, and also you can keep the destructuring in line 4 as it is.

- One more thing to note here, is that you are going to need much more logic implemented here in order to make the calculator work, but it will make much more sense in the next milestone when you integrate this logical section into the UI.